st2common/st2common/models/db/trace.py

```diff
@@ -0,0 +1,87 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import mongoengine as me
+
+from st2common.models.db import stormbase
+from st2common.fields import ComplexDateTimeField
+from st2common.util import date as date_utils
+
+from st2common.models.db import MongoDBAccess
+
+__all__ = [
+    'TraceDB',
+    'TraceComponentDB'
+]
+
+
+class TraceComponentDB(me.EmbeddedDocument):
+    """
+    """
+    object_id = me.StringField()
+    updated_at = ComplexDateTimeField(
+        default=date_utils.get_datetime_utc_now,
+        help_text='The timestamp when the TraceComponent was included.')
+
+    def __str__(self):
+        return 'TraceComponentDB@(object_id:{}, updated_at:{})'.format(
+            self.object_id, self.updated_at)
+
+
+class TraceDB(stormbase.StormFoundationDB):
```

Comment:
I guess it is a difference of expectation really. If you see the discussion I did not view Trace as a directional graph. However, I think that is the right perception. We have some of that information stored in the ActionExecution still a Trace would be ideal to store that connection information.

Right now I am trying to get out 2 usecases -
1. What are the activities performed when a Trigger was fired into the system using an external identifier?
2. What are the logical group of connected activities?

What I have provides for those cases and connections would be next on this list for sure. (How are these activities linked?) Having said that I think that will be the next level of feature set for Traces. 
The TraceComponentDB can easily be extended to provide this information therefore I am not too worried about making the change in a subsequent PR.